src/component/plotly-graph/parse.js

```diff
@@ -108,4 +112,99 @@ function parseDim (result, opts, dim) {
   }
 }
 
+function willFigureHang (result) {
+  const data = result.figure.data
+
+  // cap the number of traces
+  if (data.length > 200) {
+    return true
+  }
+
+  const numberOfPtsPerTraceType = {}
+
+  for (let i = 0; i < data.length; i++) {
+    const trace = data[i] || {}
+    const type = trace.type || 'scatter'
+
+    if (!(type in numberOfPtsPerTraceType)) {
+      numberOfPtsPerTraceType[type] = 0
+    }
+
+    numberOfPtsPerTraceType[type] += estimateDataLength(trace)
+
+    // cap the number of point per trace type
+    if (numberOfPtsPerTraceType[type] > maxPtsPerTraceType(type)) {
+      return true
+    }
+
+    // other special cases
+
+    // mesh3d will `alphahull` and 1000+ pts
+    if (
+      type === 'mesh3d' &&
+      'alphahull' in trace && Number(trace.alphahull) >= 0 &&
+      findMaxArrayLength(trace) > 1000
+    ) {
+      return true
+    }
+  }
+}
+
+// Consider the array of maximum length as a proxy to determine
+// the number of points to be drawn. In general, this estimate
+// can be (much) smaller than the true number of points plotted
+// when it does not match the length of the other coordinate arrays.
+function findMaxArrayLength (cont) {
+  const lengths = Object.keys(cont)
+    .filter(k => Array.isArray(cont[k]))
+    .map(k => cont[k].length)
+
+  return Math.max(...lengths)
+}
+
+function estimateDataLength (trace) {
+  // special case for e.g. parcoords and splom traces
+  if (Array.isArray(trace.dimensions)) {
+    return trace.dimensions
+      .map(findMaxArrayLength)
+      .reduce((a, v) => a + v)
+  }
+
+  return findMaxArrayLength(trace)
+}
+
+function maxPtsPerTraceType (type) {
+  switch (type) {
+    case 'scattergl':
+    case 'splom':
```

Comment:
Not that we really have this info yet, but I'd have thought `splom` would need to be limited quite a bit lower, perhaps the same as `parcoords` provisionally but potentially even lower.